src/WebJobs.Script.WebHost/Controllers/AdminController.cs

```diff
@@ -59,8 +62,12 @@ public HttpResponseMessage Invoke(string name, [FromBody] FunctionInvocation inv
         [Route("admin/functions/{name}/status")]
         public FunctionStatus GetFunctionStatus(string name)
         {
-            FunctionStatus status = new FunctionStatus();
             Collection<string> functionErrors = null;
+            FunctionDescriptor function = _scriptHostManager.Instance.Functions.FirstOrDefault(p => p.Name.ToLowerInvariant() == name.ToLowerInvariant());
+            FunctionStatus status = new FunctionStatus
+            {
+                Metadata = function?.Metadata
```

Comment:
I've made a simple refactoring to expose a new public ScriptHost.ReadFunctionMetadata function that you can call (see PR and example in test here https://github.com/Azure/azure-webjobs-sdk-script/pull/1117). This was easy and is preferable I think than adding new http surface area for this.